### DISCUSSION TITLE
Lighthouse deck compatible with PiCar-S?

### DISCUSSION THREAD

**Discussion by basti11699** [bitcraze/discussions#753] [Q&A]
Hi there, 
this might be quite a simple question, but is it possible to use a lighthouse deck in conjunction with a Sunfounder PiCar-S?
And if so is this possible without any major mods?

So far we have managed to attach the lighthouse deck to the PiCar, but for some unknown reason the lighthouse deck on the PiCar is not being found by our lighthouse system. All other decks installed on our crazyflie drones work perfectly fine within the same system...

Our goal is to send the lighthouse location of the PiCar carrying a small parcel (once it detects a stop sign), which should send its location to a swarm of drones, which in return should fly to the exact location of the PiCar and pick up a small parcel from the PiCar. 
Has anyone got any ideas how one might be able to setup the lighthouse deck to achieve this?

Thanks in advance. ;) #683 #77

**Comment by krichardsson** [comment]
Hi,
As far as I understand, the piCar is based on an Arduino and unfortunately this will not work easily with the Lighthouse deck. The Lighthouse deck is only doing very low level decoding of the lighthouse signals, while all higher level tasks (computing angles, calibration, geometry, position estimation) is done in the Crazyflie. There might have been some attempts of implementing this on an arduino, but I'm not aware of the status. 

The easiest solution would probably be to mount the deck on a Crayzflie based platform (Crazyflie 2.1 or Crazyflie Bolt) and put that on the car. This could also be used to communicate with the other Crazyflies. If needed, it should be possible to communicate between the Arduino and Crazyflie via a UART for instance (be ware of voltages though!).

> but for some unknown reason the lighthouse deck on the PiCar is not being found by our lighthouse system

The lighthouse base stations are only transmitting light that is received by the deck, there is no communication going back. I'm not sure which parts you refer to by "our lighthouse system", but for the Crazyflie most of the work is done in the Crazyflie, the client is mainly only showing data from the Crazyflie.
